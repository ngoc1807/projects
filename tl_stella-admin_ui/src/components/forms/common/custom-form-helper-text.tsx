import { FormHelperText, useTheme } from "@mui/material";
import { FC, useEffect, useState } from "react";

export type FormHelperTextVariants = "error" | "success" | "info";

interface Props {
  text?: string;
  variant?: FormHelperTextVariants;
  persist?: boolean;
}

export const CustomFormHelperText: FC<Props> = ({
  text = " ",
  variant = "info",
  persist = true,
}) => {
  const theme = useTheme();
  const successStyle = { color: theme.palette.success.dark };

  const [formHelperText, setFormHelperText] = useState(" " as string);

  useEffect(() => {
    setFormHelperText(text);
    if (!persist) {
      setTimeout(() => {
        setFormHelperText(" ");
      }, 3500);
    }
  }, [text, persist]);

  return (
    <FormHelperText
      error={variant === "error"}
      sx={
        variant === "success"
          ? { minHeight: "11px", ...successStyle }
          : { minHeight: "11px" }
      }
    >
      {formHelperText}
    </FormHelperText>
  );
};
