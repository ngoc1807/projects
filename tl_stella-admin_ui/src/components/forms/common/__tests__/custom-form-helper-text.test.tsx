import { cleanup, render } from "@testing-library/react";
import { CustomFormHelperText } from "../custom-form-helper-text";

const errorHelperText = "Oh no there's an error!";
const infoHelperText = "Informational helper";
const successHelperText = "Success!";

describe("custom-form-helper-text", () => {
  afterEach(() => {
    cleanup();
  });
  it("should render correctly when only required props are passed", () => {
    const { container } = render(<CustomFormHelperText />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it("should render correctly when optional props are passed for informational messaging", () => {
    const { container } = render(
      <CustomFormHelperText
        persist={false}
        variant="info"
        text={infoHelperText}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it("should render correctly when optional props are passed for success messaging", () => {
    const { container } = render(
      <CustomFormHelperText
        persist={true}
        variant="success"
        text={successHelperText}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  it("should render correctly when optional props are passed for error messaging", () => {
    const { container } = render(
      <CustomFormHelperText
        persist={true}
        variant="error"
        text={errorHelperText}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
