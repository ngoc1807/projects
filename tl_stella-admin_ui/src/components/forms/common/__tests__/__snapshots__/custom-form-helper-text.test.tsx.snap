// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`custom-form-helper-text should render correctly when only required props are passed 1`] = `
<p
  class="MuiFormHelperText-root css-pxg7nu-MuiFormHelperText-root"
>
  <span
    class="notranslate"
  >
    â€‹
  </span>
</p>
`;

exports[`custom-form-helper-text should render correctly when optional props are passed for error messaging 1`] = `
<p
  class="MuiFormHelperText-root Mui-error css-pxg7nu-MuiFormHelperText-root"
>
  Oh no there's an error!
</p>
`;

exports[`custom-form-helper-text should render correctly when optional props are passed for informational messaging 1`] = `
<p
  class="MuiFormHelperText-root css-pxg7nu-MuiFormHelperText-root"
>
  Informational helper
</p>
`;

exports[`custom-form-helper-text should render correctly when optional props are passed for success messaging 1`] = `
<p
  class="MuiFormHelperText-root css-f1s83x-MuiFormHelperText-root"
>
  Success!
</p>
`;
